Gems/PhysX/Code/Tests/PhysXScriptTest.cpp

```diff
@@ -153,4 +153,385 @@ namespace PhysX
 
         EXPECT_TRUE(GetScriptContext()->Execute(luaCode));
     }
+
+    TEST_F(PhysXScriptTest, SceneRayCast_RaycastNotIntersectingBox_ReturnsNoHits)
+    {
+        s_testEntities.insert(
+            {
+                "Box",
+                TestUtils::AddStaticUnitTestObject<BoxColliderComponent>(GetDefaultSceneHandle(), AZ::Vector3::CreateZero(), "Box")
+            });
+
+        const char luaCode[] =
+            R"(
+                physicsSystem = GetPhysicsSystem()
+                sceneHandle = physicsSystem:GetSceneHandle(DefaultPhysicsSceneName)
+                scene = physicsSystem:GetScene(sceneHandle)
+                request = RayCastRequest()
+                request.Start = Vector3(5.0, 0.0, 5.0)
+                request.Direction = Vector3(0.0, 0.0, -1.0)
+                request.Distance = 10.0
+                hits = scene:QueryScene(request)
+                ExpectTrue(hits.HitArray:Size() == 0)
+            )";
+
+        EXPECT_TRUE(GetScriptContext()->Execute(luaCode));
```

Comment:
Is this only checking that the script gets executed? Will `Execute` return false if the scripts' `ExpectTrue` fails?

Same in the other tests.

Reply:
I think that just checks that the script has run without any script errors. It will fail the test inside ExpectTrue if the test logic itself fails. I have verified that it will fail if you have e.g. ExpectTrue(false) in side the script.

Reply:
Perfect, thanks.